src/openapi/streaming/connection/transport/signalr-core-transport.js

```diff
@@ -334,58 +330,84 @@ SignalrCoreTransport.prototype.updateQuery = function(
-    const options = {
-        headers: {
-            Authorization: authToken,
-        },
-    };
-    const url = `${this.baseUrl}/streaming/renewal/renewsession`;
+    if (!this.connection || this.isReconnecting) {
+        return;
+    }
 
-    return fetch('POST', url, options)
-        .then(() => {
-            log.debug(
-                LOG_AREA,
-                'Streaming session is successfully renewed with new token',
-                {
-                    contextId,
-                },
-            );
+    return this.connection
+        .invoke('RenewToken', authToken)
+        .then(({ Status }) => {
+            switch (Status) {
+                case renewStatus.SUCCESS:
+                    log.info(
+                        LOG_AREA,
+                        'Successfully renewed token for session',
+                        {
+                            contextId,
+                        },
+                    );
+                    return;
+
+                case renewStatus.INVALID_TOKEN:
+                    // if this call was superseded by another one, then ignore this error
+                    // else let auth provider know of invalid token
+                    if (this.authToken === authToken) {
+                        this.unauthorizedCallback();
+                    }
+
+                    return;
+
+                case renewStatus.SESSION_NOT_FOUND:
+                    log.warn(
+                        LOG_AREA,
+                        'Session not found while renewing session',
+                        {
+                            contextId,
+                        },
+                    );
+
+                    // should try to reconnect with different context id
+                    this.stop();
+                    return;
+
+                default:
+                    log.error(
+                        LOG_AREA,
+                        'Unknown status received while renewing session',
+                        {
+                            status: Status,
+                            contextId,
+                        },
+                    );
+
+                    this.stop(true);
+                    this.transportFailCallback();
+            }
         })
         .catch((error) => {
-            // if this call was superseded by another one, then ignore this error
-            if (this.authToken !== authToken) {
+            // Invocation could get cancelled due to connection being closed
+            if (!this.connection || this.isReconnecting) {
                 return;
             }
 
-            // if a network error occurs, retry
-            if (error && error.isNetworkError) {
-                return this.renewSession(authToken, contextId);
-            }
-
-            if (error.status === 401) {
-                this.unauthorizedCallback();
-                return;
-            }
-
-            log.error(
+            log.info(

```

Comment:
should this be a warn or error? it is an unexpected scenario?

Reply:
this could occur when signalr lib is unable to send websocket message
it could be due to network error
I don't see it as unexpected scenario 
other scenarios could be streaming service is not available and in that case connection will be dropped

Reply:
I will change it to warning

Reply:
will change it to warning